### DISCUSSION TITLE
What are the breaking changes in 3.0.0?

### DISCUSSION THREAD

**Discussion by SubJunk** [jupnp/jupnp#224] [Q&A]
Thanks for the new release! Are there any known breaking changes? It wasn't clear to me in the release notes

**Comment by wborn** [comment]
The most breaking change is that it requires Java 11 or newer now instead of Java 7.
The mocks also moved into a test JAR (https://github.com/jupnp/jupnp/pull/181, https://github.com/jupnp/jupnp/pull/198).
It also no longer supports the deprecated SecurityManager (https://github.com/jupnp/jupnp/pull/168).
Besides that I don't know of anything else breaking after testing it with openHAB.

**Comment by wborn** [comment]
I've now also added this info to the [3.0.0 release](https://github.com/jupnp/jupnp/releases/tag/3.0.0).
